src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -276,7 +276,7 @@ public override bool Equals(object? obj)
 
         // Track some of the persisted path hash. Used to be able to remove keys when it is replaced.
         // If null, disable removing key.
-        private LruCacheLowObject<HashAndTinyPath, ValueHash256>? _pastPathHash;
+        private LruCache<HashAndTinyPath, ValueHash256>? _pastPathHash;
```

Comment:
Regular LruCache drops block validation throughput by 2/3rds after second prune 😢

Reply:
Can you try `--Db.StateDbWriteBufferSize 256000000` and see if it take about 8 to 10 prune to happen again?

Reply:
Will tell you the result in many hours 😅 The prunes aren't that frequent

Reply:
this would mean something is wrong with LruCacheLowObject

Reply:
@benaadams can you reduce pruning cache to small number so will be more frequent?

Reply:
Well, the memory pruning time increase, so thats unfortunate.

Reply:
> Can you try --Db.StateDbWriteBufferSize 256000000 and see if it take about 8 to 10 prune to happen again?

Seems to have not really slowed down yet; the is a drift down and a drift up in memory; but nothing as dramatic as before
<img width="758" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/7ac047fa-6137-4dca-9e62-770e670fcfca">

> That is likely the reason you observe slowdown instead of this variable itself.

What's the premise here? The change caused less items to be deleted, so more new files to be written? Which means it didn't get slower? 🤔

Reply:
Less new files. Delete takes up space until it is compacted to bottom.

Reply:
> see if it take about 8 to 10 prune to happen again?

Small drop at 5; large drop at 15
![image](https://github.com/NethermindEth/nethermind/assets/1142958/9a03fc9f-3703-4f17-b3be-c8129544ec8e)

Reply:
Then the drop is not because of the LRU, but likely about creation of new uncached files.

Reply:
Any way to resolve?

Reply:
The weird bit is restarting (without the file preload) and performance jumps back up

Reply:
No idea. Maybe try on linux to see if it has the same effect.

Reply:
Did a manual db load; as done on startup, didn't change anything.

Would `WriteFlags.DisableWAL` for the millions of removes help?

Reply:
Or flush the db cache if its now filled with lots of stuff that will no longer be accessed?

Reply:
Bypassing WAL for the delete batch seems to have fixed it; testing a bit more